src/sentry/static/sentry/app/views/settings/components/dataScrubbing/index.tsx

```diff
@@ -75,7 +75,11 @@ class DataScrubbing<T extends ProjectId = undefined> extends React.Component<
     this.loadOrganizationRules();
   }
 
-  componentDidUpdate(_prevProps: Props<T>, prevState: State) {
+  componentDidUpdate(prevProps: Props<T>, prevState: State) {
+    if (prevProps.relayPiiConfig !== this.props.relayPiiConfig) {
+      this.loadRelayPiiConfig();
+      return;
+    }
```

Comment:
We can avoid adding more state and lifecycle checks if we forced this component to re-mount when the project changes by adding a `key` where we render it.